player/command.c

```diff
@@ -6713,6 +6713,9 @@ void mp_option_change_callback(void *ctx, struct m_config_option *co, int flags,
 
     if (opt_ptr == &opts->vo->taskbar_progress)
         update_vo_playback_state(mpctx);
+
+    if (opt_ptr == &opts->watch_later_properties)
+        mp_get_resume_defaults(mpctx);
```

Comment:
if runtime change support is unnecessary, please remove all relative codes. It doesn't seem to work as expected because the current option value may not be the default value.

Reply:
It is? If we didn't want runtime support we could have used the simpler version. It's documented in the manpage that the value at the time options are added is used. The alternative is storing a copy of every option and property on startup and finding it for each proeprty in `mp_write_watch_later_conf`, which is slower, though we can try that too.

Reply:
as I said, we cannot prevent user from changing it only through the man page. The name-value list works regardless of whether the user modifies it or not, while the "simpler version" does not. "no runtime change support" means that we allow user to change it but ignore the changes.

Reply:
I don't understand. This supports runtime changes in the way documented in the manpage. And this name-value list doesn't work if when the user modifies `watch-later-properties` at runtime `mp_get_resume_defaults` isn't called again since `mp_write_watch_later_conf` still does the comparison based on the array indexes.